### PR TITLE
Add `SortDocumentSymbols` to make the outline hierarchical (again)

### PR DESCRIPTION
**Pull request by andyleejordan**
This is probably what OmniSharp was doing for us. Fixes the bug introduced by #2083.

Hey, at least I didn't use LINQ. Thanks Bing Chat for the algorithm help. This should probably wait for the pre-release after the next stable release.

### CODE REVIEW

**Review by andyleejordan — COMMENTED** [review]
Feedback from code review meeting.

File: src/PowerShellEditorServices/Services/TextDocument/Handlers/DocumentSymbolHandler.cs [hunk 1]
```diff
@@ -40,15 +40,106 @@ protected override DocumentSymbolRegistrationOptions CreateRegistrationOptions(D
...
+            }
+
+            // Now recursively sort the children into nested buckets of children too.
+            foreach (HierarchicalSymbol parent in parents)
+            {
+                List<HierarchicalSymbol> sortedChildren = await SortHierarchicalSymbols(parent.Children, cancellationToken).ConfigureAwait(false);
+                // Since this is a foreach we can't just assign to parent.Children and have to do
+                // this instead.
+                parent.Children.Clear();
+                parent.Children.AddRange(sortedChildren);
```
Comment (andyleejordan) [inline src/PowerShellEditorServices/Services/TextDocument/Handlers/DocumentSymbolHandler.cs:96]: Is there a good way to optimize this? @SeeminglyScience

**Review by SeeminglyScience — APPROVED** [review]
LGTM! Love it ❤️

**Review by SeeminglyScience — APPROVED** [review]
Even better ❤️

**Comment by Stephanevg** [comment]
Hi @andyleejordan thank you so much for working on this bug! I see the status is now merged in main, does that mean that this fix is already present in the latest version of vscode? From which version ?

Because I just downloaded the latest version (see below) and the bug still exists there😿 

```
Version: 1.84.1 (system setup)
Commit: 2b35e1e6d88f1ce073683991d1eff5284a32690f
Date: 2023-11-06T12:35:59.022Z
Electron: 25.9.2
ElectronBuildId: 24603566
Chromium: 114.0.5735.289
Node.js: 18.15.0
V8: 11.4.183.29-electron.0
OS: Windows_NT x64 10.0.22000
```

Or maybe I am just being to impatient.... 🙄 😊 ?
Thanks again!

**Pull request merged**